internal/proxy/validate_util.go

```diff
@@ -35,6 +35,37 @@ func isNumber(c uint8) bool {
 	return true
 }
 
+func ValidateCollectionAlias(collAlias string) error {
+	collAlias = strings.TrimSpace(collAlias)
```

Comment:
why do we need ValidateCollectionAlias, thought the rule should be same as Validate CollectionName

Reply:
I thought about using the same function, but the error messages will be confusing.

Reply:
That's a good point, maybe we can do
if !isValueCollectionName(colAlias) {
   return fmt.Error("xxx")
}
isValueCollectionName can be shared

Reply:
There are 4 types of error causes, and it is hard to achieve reusability this way.  I've singled out a helper function to do the actual validation.

Reply:
```
func validateCollectionNameOrAlias(entity, entityType string) error {
	entity = strings.TrimSpace(entity)

	if entity == "" {
		return errors.New(fmt.Sprintf("Collection %s should not be empty", entityType))
	}

	invalidMsg := fmt.Sprintf("Invalid collection %s: %s. ", entityType, entity)
	if int64(len(entity)) > Params.MaxNameLength {
		msg := invalidMsg + fmt.Sprintf("The length of a collection %s must be less than ", entityType) +
			strconv.FormatInt(Params.MaxNameLength, 10) + " characters."
		return errors.New(msg)
	}

	firstChar := entity[0]
	if firstChar != '_' && !isAlpha(firstChar) {
		msg := invalidMsg + fmt.Sprintf("The first character of a collection %s must be an underscore or letter.", entityType)
		return errors.New(msg)
	}

	for i := 1; i < len(entity); i++ {
		c := entity[i]
		if c != '_' && c != '$' && !isAlpha(c) && !isNumber(c) {
			msg := invalidMsg + fmt.Sprintf("Collection %s can only contain numbers, letters, dollars and underscores.", entityType)
			return errors.New(msg)
		}
	}
	return nil
}
```